pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -359,8 +362,10 @@ func (ds *dockerService) getIP(podSandboxID string, sandbox *dockertypes.Contain
 	return ""
 }
 
-// PodSandboxStatus returns the status of the PodSandbox.
-func (ds *dockerService) PodSandboxStatus(ctx context.Context, req *runtimeapi.PodSandboxStatusRequest) (*runtimeapi.PodSandboxStatusResponse, error) {
+// internalPodSandboxStatus returns the status of the PodSandbox, optionally
+// ignoring the sandbox IP address since that can depend on things docker may
+// have already torn down while stopping sandbox (eg, Linux network namespace).
+func (ds *dockerService) internalPodSandboxStatus(ctx context.Context, req *runtimeapi.PodSandboxStatusRequest, checkIP bool) (*runtimeapi.PodSandboxStatusResponse, error) {
```

Comment:
StopPodSandbox() only needs to know the pod's name and namespace, and whether it's using hostNetwork. Would it make sense to just have a smaller status function to just get that information? (Or even to inline the calls into StopPodSandbox.)

Reply:
It would be nice if we can refactor  PodSandboxStatus a bit for this. But probably means more work...